solutions/swb-reference/integration-tests/support/resources/environmentTypeConfigs/environmentTypeConfigs.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import ClientSession from '../../clientSession';
+import RandomTextGenerator from '../../utils/randomTextGenerator';
+import CollectionResource from '../base/collectionResource';
+import EnvironmentTypeConfig from './environmentTypeConfig';
+
+export default class EnvironmentTypeConfigs extends CollectionResource {
+  private _parentId: string;
+  public constructor(clientSession: ClientSession, parentApi: string, parentId: string) {
+    super(clientSession, 'environmentTypeConfigs', 'environmentTypeConfig');
+    this._api = `${parentApi}/configurations`;
+    this._parentId = parentId;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  public environmentTypeConfig(id: string): EnvironmentTypeConfig {
+    return new EnvironmentTypeConfig(id, this._clientSession, this._api, this._parentId);
+  }
+
+  protected _buildDefaults(resource: ETCCreateRequest): ETCCreateRequest {
+    const randomTextGenerator = new RandomTextGenerator(this._settings.get('runId'));
+    return {
+      description: resource.description ?? randomTextGenerator.getFakeText('fakeDescription'),
+      name: resource.name ?? randomTextGenerator.getFakeText('fakeName'),
+      type: resource.type ?? randomTextGenerator.getFakeText('fakeType'),
+      params: resource.params ?? [],
+      estimatedCost: resource.name ?? randomTextGenerator.getFakeText('fakeEstimatedCost')
+    };
+  }
+}
+
+interface ETCCreateRequest {
```

Comment:
Do you still need this if we have the zod parser type?

Reply:
this only gets the defaults which doesn't necessarily match the create request object, in this case they don't match , I could use the .pick from Zod and infer another type but IMO its easier to read this way